src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/HMPPSAuthService.kt

```diff
@@ -44,6 +70,38 @@ class HMPPSAuthService(
     return if (serviceProviderOrgs.isEmpty()) null else serviceProviderOrgs[0]
   }
 
+  fun getUserDetail(user: AuthUser): UserDetail {
+    return if (user.authSource == "auth") {
+      val url = UriComponentsBuilder.fromPath(authUserDetailLocation).buildAndExpand(user.userName).toString()
+      hmppsAuthApiWebClient.get().uri(url)
+        .retrieve()
+        .bodyToMono(AuthUserDetailResponse::class.java)
+        .map {
+          if (!it.verified) {
+            throw UnverifiedEmailException()
```

Comment:
Could we validate that the person has a verified e-mail address in other control flows? This (`getUserDetail`) seems like an unusual place to do it

Reply:
i don't fully understand the 403 idea. but the thinking here is that the caller of this method wants the user details, including the email address. if the email address is unverified, then the method blows up because the email address isn't reliable. so it seems like an ok place to check this to me.

Reply:
Sure, sorry, I meant: should we explicitly say no one with an unverified email can access or be assigned in our service?

Reply:
ok i see what you mean.

first off this is just a method to get user details for a username. so i don't think we can make any assumptions about whether the username has a verified email in this class. 

second, we will do the validation you are ttalking about on the front end anyway. we could do it in the API as well, but i still don't think it changes the fact that this method should be agnostic and handle unverified emails. it just means these UnverifiedEmailExceptions should never be thrown :) !